exporter/clickhousetracesexporter/clickhouse_factory.go

```diff
@@ -78,54 +77,64 @@ func ClickHouseNewFactory(migrations string, datasource string, dockerMultiNodeC
 	}
 }
 
+// Round-robin connection-selection strategy.
+func (f *Factory) selectConn() clickhouse.Conn {
+	if len(f.conns) > 1 {
+		f.conns = append(f.conns[1:], f.conns[0])
+	}
+	return f.conns[0]
+}
+
 // Initialize implements storage.Factory
-func (f *Factory) Initialize(logger *zap.Logger) error {
+func (f *Factory) Initialize(logger *zap.Logger) (clickhouse.Conn, error) {
```

Comment:
Is there a clear way in Go to indicate _what_ this return-value is, semantically? (I name the return-value `init_conn` below, but idk if folks are going to reflexively check the source …)

I'd rather indicate in the type/signature that the user shouldn't use the returned `conn` for anything other than initialization …